src/charm.py

```diff
@@ -133,6 +133,28 @@ def set_pod_spec(self, event):
                     }
                 ],
             },
+            k8s_resources={
+                "kubernetesResources": {
+                    "services": [
+                        {
+                            "name": "metadata-grpc-service",
+                            "spec": {
+                                "selector": {
+                                    "app.kubernetes.io/name": self.model.app.name
+                                },
+                                "ports": [
+                                    {
+                                        "name": "grpc-api",
+                                        "port": int(config["port"]),
```

Comment:
I still don't like that we are treating `port` as a string, with SDI this shouldn't be an issue anymore.